user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md

```diff
@@ -0,0 +1,118 @@
+  - *UserDefinableApiEndpointInstaller.txt* contains logs from the installation of the DxM if it is installed by a DataMiner upgrade package or via the command line. This will not contain additional logs when you use the dedicated installer for the DxM. A small subset of these log lines will also be available in the *progress.log* DataMiner upgrade package log file and will be displayed during the upgrade.
+- The file *SLUserDefinableApiManager.txt* in the `C:\Skyline DataMiner\Logging\` folder contains logging related to the manager in SLNet that manages the CRUD actions on API tokens/definitions and the API triggers.
+
+> [!NOTE]
+> When there are more than two Agents in a DMS, the API triggers arriving on an endpoint are forwarded randomly to one of the DataMiner Agents. Keep this in mind when checking the logging. If an HTTP request was sent to the endpoint running on a specific server, there is no guarantee that this trigger will be handled by the DataMiner Agent on that same server. The JSON error responses returned by the endpoint should include a *faultingNode* field that indicates the ID of the DMA that actually handled the request.
+
+## Information events
+
+An information event will be generated for every create, update, and delete action for an API token or definition. These information events include the following details:
+
+- **API token**: The name, the token ID, and the user who executed the action.
+- **API definition**: The URL route, the definition ID, and the user who executed the action.
+
+## Request flow
+
+Below, you can find the request flow when a trigger comes in. This may help you during troubleshooting.
+
+```mermaid
+flowchart LR
+client("User") -- "HTTP(S) request" --> a
+subgraph "IIS"
+    a("IIS") --> b("Rewrite Rule")
+end
+subgraph "Extension"
+    b -- HTTP request --> c("UserDefinableApiEndpoint\nDxM")
+end
+subgraph "Core DataMiner"
+    c -- NATS --> d("UserDefinableApiEndpointManager\n(SLNet.exe)") 
+    d -- IPC --> e("SLAutomation.exe")
+end
+```
+
+## Installation issues
+
+The errors in this section can be generated when the UserDefinableApiEndpoint DxM is installed via a DataMiner upgrade package. These will be visible in the upgrade log during the upgrade and also in the `UserDefinableApiEndpoint.txt` log file as mentioned above under [Logging](#logging).
+
+- **Could not find AspNetCore installed on this system**
+
+  DataMiner should install ASP.NET Core 5 automatically, but in case this failed or in case it has been removed, the installer can generate this error.
+
+  To fix this:
+
+  1. Go to the [.NET 5 download page](https://dotnet.microsoft.com/en-us/download/dotnet/5.0) and download the latest ASP.NET Core Runtime 5.
+
+  1. On the DMA, got to the folder `C:\Skyline DataMiner\Tools\ModuleInstallers\` and run the installer `DataMiner UserDefinableApiEndpoint 1.X.X.X.msi`.
+
+- **Found rewrite rule, but has inconsistencies:**
```

Comment:
Checked the source code, it should be "Found rewrite rule, but it has inconsistencies:"